src/pages/permissions/CreateIdentityModal.tsx

```diff
@@ -0,0 +1,127 @@
+import { FC, useState } from "react";
+import {
+  ActionButton,
+  Button,
+  Icon,
+  Input,
+  Modal,
+  Notification,
+} from "@canonical/react-components";
+import { useFormik } from "formik";
+import { createFineGrainedTlsIdentity } from "api/auth-identities";
+import { base64EncodeObject } from "util/helpers";
+import { useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+
+interface Props {
+  onClose: () => void;
+}
+
+const CreateIdentityModal: FC<Props> = ({ onClose }) => {
+  const [token, setToken] = useState<string | null>(null);
+  const [error, setError] = useState(false);
+  const queryClient = useQueryClient();
+
+  const formik = useFormik({
+    initialValues: {
+      identityName: "",
+    },
+    onSubmit: (values) => {
+      createFineGrainedTlsIdentity(values.identityName)
+        .then((response) => {
+          const encodedToken = base64EncodeObject(response);
+          setToken(encodedToken);
+
+          void queryClient.invalidateQueries({
+            queryKey: [queryKeys.identities],
+          });
+        })
+        .catch(() => {
+          setError(true);
+        });
+    },
+  });
+
+  return (
+    <Modal
+      close={onClose}
+      className="create-tls-identity"
+      title={token ? "Identity has been created" : "Generate trust token"}
+      buttonRow={
+        <>
+          {token && (
+            <Button
+              aria-label="Copy token"
+              title="Copy token"
+              className="u-no-margin--bottom"
+              onClick={() => {
+                void navigator.clipboard.writeText(token);
+              }}
+              type="button"
+              hasIcon
+            >
+              <Icon name="copy" />
+            </Button>
```

Comment:
Can we add some feedback after this is used?
A [good example interaction is found in AMS](https://github.com/canonical/ams/blob/main/ui/templates/js/components/ui/CopyToClipboard.tsx#L10):